src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/framework/api/aggregators/PersistableSlidingWindow.java

```diff
@@ -94,5 +110,6 @@ protected synchronized void write() throws IOException {
     writer.flush();
```

Comment:
You need to close the stream here after flushing. You can use a try-with-resources block so that it gets autoclosed after using.

Reply:
Good catch, done